src/main/java/de/tum/in/www1/artemis/repository/ProgrammingExerciseRepository.java

```diff
@@ -73,8 +73,8 @@ public interface ProgrammingExerciseRepository extends JpaRepository<Programming
     @EntityGraph(type = LOAD, attributePaths = { "templateParticipation", "solutionParticipation" })
     Optional<ProgrammingExercise> findWithTemplateAndSolutionParticipationById(long exerciseId);
 
-    @EntityGraph(type = LOAD, attributePaths = { "categories", "teamAssignmentConfig", "templateParticipation.submissions.results", "solutionParticipation.submissions.results",
-            "auxiliaryRepositories" })
+    @EntityGraph(type = LOAD, attributePaths = { "categories", "teamAssignmentConfig", "templateParticipation.results", "templateParticipation.submissions.results",
+            "solutionParticipation.results", "solutionParticipation.submissions.results", "auxiliaryRepositories" })
     Optional<ProgrammingExercise> findWithTemplateAndSolutionParticipationSubmissionsAndResultsAndAuxiliaryRepositoriesById(long exerciseId);
```

Comment:
De we really need all of these files everywhere where this method is used? If we only need them for on case, add an additional method to avoid overhead in other cases. But in general I don't see why we should need the results twice.